doc/adr/0002-promote-tenant-cmd.md

```diff
@@ -0,0 +1,41 @@
+# 1. Corectl promote-image command
+
+Date: 2024-07-22
+
+## Status
+
+Accepted
+
+## Context
+
+We want a way to promote docker images between environments.
+Currently, we implement this via Makefile tasks which means that we have duplicate code to every application 
+repository. To reduce the amount of duplicate code, we want to move it to a `corectl` command
+
+## Command interface
+
+TODO: authentication arguments
+
+### Option 1
+
+This option requires operator to know details about the cloud environment, i.e `gcp projectId`
+
+```shell
+corectl p2p promote-image 
+    --imageUri=${gcp_region}-docker.pkg.dev/${gcp_project_id}/tenant/${tenant}/${stage}/lukasz-app-1:0.0 \ 
+    --to-registry=${gcp_region}-docker.pkg.dev/${gcp_project_id}/tenant/${tenant}/${stage}/lukasz-app-1
+```
+
+### Option 2
+
+We can rely on `corectl` to get the details about the cloud environment. 
+
+```shell
+corectl p2p promote-image 
+    --image=lukasz-app-1:0.0 \
+    --tenant=lukasz \
+    --from-env=gcp-dev \
+    --from-stage=fast-feedback \
```

Comment:
It would be great if we could infer the environment based on the stage. I haven't seen any mapping, is there one?

Reply:
you should be able to infer this since you have this on env vars `FAST_FEEDBACK: {"include":[{"deploy_env":"gcp-dev"}]}`
maybe we maybe just have
```
corectl p2p promote-image 
    --image=lukasz-app-1:0.0 \
    --tenant=lukasz \
    --to=extended-test    
    ```
    and infer the rest? this would mean it follows a p2p and pushes tot he correct environments. I think this also implies we'll need to have a push-to tasks done by corectl

Reply:
> you should be able to infer this since you have this on env vars `FAST_FEEDBACK: {"include":[{"deploy_env":"gcp-dev"}]}`

@tmcalves sweet, I think this is a good starting point then

```
corectl p2p promote-image 
    --image=lukasz-app-1:0.0 \
    --tenant=lukasz \
    --from=fast-feedback \
    --to=extended-test 
```
otherwise, I would have to either hardcode p2p path or add ability to configure it in corectl 

---
> Wondering if when you connect to environments you can omit details like --from-env which it will already infer. Similar with tenant picker

@SennaSemakula I think infering env from github variable as @tmcalves suggested will work well

Reply:
I think even more optionated

corectl p2p promote --type=image (for future proofing) --from=fastfeedback

There isn't an option to go from fast feedback to prod say, or extended test to fastfeedback, so lets not five it to them

Reply:
should we keep the `to` rather than the `from`?  I as tenant I care more about that the image is in the env that I want and less where it's coming from

Reply:
just thinking about how it will look in the P2P

The last stage of P2P in fastfeedback is promote, so you know where you're at

I don't feel strongly eithe rway

Reply:
I still need to pass an image though, right? `p2p promote image:tag --from fastfeedback`